examples/images_contours_and_fields/image_annotated_heatmap.py

```diff
@@ -0,0 +1,313 @@
+"""
+===========================
+Creating annotated heatmaps
+===========================
+
+It is often desireable to show data which depends on two independent
+variables as a colorcoded image plot. This is often referred to as a
+heatmap. If the data is categorical, this would be called a categorical
+heatmap.
+Matplotlib's :meth:`imshow <matplotlib.axes.Axes.imshow>` function makes
+production of such plot particularly easy.
+
+The following examples show how to create a heatmap with annotations.
+We will start with an easy example and expand it to be usable as a
+universal function.
+"""
+
+# sphinx_gallery_thumbnail_number = 2
+
+##############################################################################
+#
+# A simple categorical heatmap
+# ----------------------------
+#
+# We may start by defining some data. What we need is a 2D list or array
+# which defines the data to color code. We then also need two lists or arrays
+# of categories; of course the number of elements in those lists
+# need to match the data along the respective axes.
+# The heatmap itself is an :meth:`imshow <matplotlib.axes.Axes.imshow>` plot
```

Comment:
```
`~.Axes.imshow`
```
Will give you the same thing I am 98% sure as this long string (including the `:meth:`)